analyzer.go

```diff
@@ -333,12 +333,28 @@ func (gosec *Analyzer) load(pkgPath string, conf *packages.Config) ([]*packages.
 		return []*packages.Package{}, fmt.Errorf("importing dir %q: %w", pkgPath, err)
 	}
 
-	var packageFiles []string
-	for _, filename := range basePackage.GoFiles {
-		packageFiles = append(packageFiles, path.Join(pkgPath, filename))
+	var goModFile string
+	goModDir := abspath[0 : len(abspath)-len(pkgPath)] // get root dir
+	if os.Getenv("DISABLE_MULTI_MODULE_MODE") != "true" {
```

Comment:
Doesn't the multi-module project requires a workspace file? https://go.dev/ref/mod#workspaces

The switch from single module to multi module project is controlled by GOWORK environment variable and this should be already supported inside of Load function.

Reply:
It is possible to use the replace directive in a `go.mod`. See: https://go.dev/ref/mod#go-mod-file-replace

`go.work` was introduced later in go1.18 to simplify multi-module projects but is not mandated iirc